addons/web/controllers/main.py

```diff
@@ -1613,8 +1613,10 @@ def upload_attachment(self, model, id, ufile, callback=None):
                 })
                 attachment._post_add_create()
             except Exception:
-                args.append({'error': _("Something horrible happened")})
-                _logger.exception("Fail to upload attachment %s" % ufile.filename)
+                if sys.exc_info()[0] == AccessError:
```

Comment:
Easier way to write?

```py
            except AccessError:
                args.append({'error': _("You are not allowed to upload an attachment here.")})
            except Exception:
                args.append({'error': _("Something horrible happened")})
```

Reply:
> Easier way to write?
> 
> ```python
>             except AccessError:
>                 args.append({'error': _("You are not allowed to upload an attachment here.")})
>             except Exception:
>                 args.append({'error': _("Something horrible happened")})
> ```

Both `UserErro`r and `AccessError` give the same result. So I correctly checked `sys.exc_info()[0] == AccessError`

Reply:
What does `UserError` has to do here?!

I join seb in saying that is code is code equivalent and easier to what you coded.